source/bot.ts

```diff
@@ -86,6 +89,20 @@ bot.on("voiceStateUpdate", (oldState, newState) => {
    voiceChatService.handleVoiceStateEvent(oldState, newState);
 });
 
+bot.on("guildMemberAdd", async member => 
+{
+   let autoRoleArray: Array<AutoRole> = await autoRoleService.getAllAutoRoles(member.guild);
+
+   for(let something of autoRoleArray)
+   {
+      let role: Role = await member.guild.roles.fetch(something.role);
+      if(role)
+      if(!member.roles.cache.has(role.id)) 
```

Comment:
no need to check if the member already has the role since he just joined so he cant possibly have the role